packages/@aws-cdk/aws-apprunner/lib/service.ts

```diff
@@ -0,0 +1,603 @@
+   * If specified, this command overrides the Docker image’s default start command.
+   *
+   * @default - no start command
+   */
+  readonly startCommand?: string;
+}
+
+/**
+ * Describes a source image repository.
+ *
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apprunner-service-imagerepository.html
+ */
+interface ImageRepository {
+  /**
+   * The identifier of an image.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apprunner-service-imagerepository.html
+   */
+  readonly imageIdentifier: string;
+  /**
+   * The type of the image repository. This reflects the repository provider and whether
+   * the repository is private or public.
+   */
+  readonly imageRepositoryType: ImageRepositoryType;
+  /**
+   * Configuration for running the identified image.
+   */
+  readonly imageConfiguration: ImageConfiguration;
+}
+
+/**
+ * Identifies a version of code that AWS App Runner refers to within a source code repository.
+ *
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apprunner-service-sourcecodeversion.html
+ */
+export interface SourceCodeVersion {
+  /**
+   * The type of version identifier.
+   */
+  readonly type: string;
+  /**
+   * A source code version.
+   */
+  readonly value: string;
+}
+
+/**
+ * Properties of the CodeRepository.
+ */
+export interface CodeRepositoryProps {
+  /**
+   * Configuration for building and running the service from a source code repository.
+   */
+  readonly codeConfiguration: CodeConfiguration;
+  /**
+   * The location of the repository that contains the source code.
+   */
+  readonly repositoryUrl: string;
+  /**
+   * The version that should be used within the source code repository.
+   */
+  readonly sourceCodeVersion: SourceCodeVersion;
+}
+
+/**
+ * Properties of the AppRunner Service
+ */
+export interface ServiceProps {
+  /**
+   * The description of a source image repository.
+   * You must provide either `image` or `code`(but not both).
+   *
+   * @default - no image repository.
+   */
+  readonly image?: ContainerImage;
+  /**
+   * The description of a source code repository.
+   * You must provide either `image` or `code`(but not both).
+   *
+   * @default - no code repository.
+   */
+  readonly code?: CodeRepositoryProps;
+  /**
+   * The port that your application listens to in the container.
+   *
+   * @default 8080
+   */
+  readonly port?: number;
+  /**
+   * The number of CPU units reserved for each instance of your App Runner service.
+   *
+   * @default - 1 vCPU
+   */
+  readonly cpu?: Cpu;
+  /**
+   * The amount of memory reserved for each instance of your App Runner service.
+   *
+   * @default - 2 GB
+   */
+  readonly memory?: Memory;
+  /**
+   * ARN of the connection to Github. Only required for Github source.
+   *
+   * @default - no connection
+   */
+  readonly connection?: Connection;
```

Comment:
This should probably be called GitHubConnection then...

```suggestion
   * ARN of the connection to Github. Only required for Github source.
   *
   * @default - no connection
   */
  readonly gitHubConnection?: GitHubConnection;
```